rs_client.py

```diff
@@ -1019,6 +1019,7 @@ def _rs_struct(typeobj):
 
         _r('pub type %s%s = base::StructPtr<%s%s>;', typeobj.rs_type, lifetime1,
                 lifetime2, typeobj.ffi_type)
+        typeobj.struct_ptr = True
 
```

Comment:
Regarding the first two points, I agree. However, what part do you mean is "not necessary"? The `hasaddr` checks?